pkg/cmd/cli/cmd/login.go

```diff
@@ -0,0 +1,250 @@
+			if userFullName, err := whoami(clientCfg); err == nil && (!options.usernameProvided() || (options.usernameProvided() && options.username == userFullName)) {
+				err = config.UpdateConfigFile(userFullName, clientCfg.BearerToken, f.OpenShiftClientConfig.KClientConfig, configStore)
+				checkErr(err)
+				fmt.Printf("Already logged into %v as '%v'\n", clientCfg.Host, userFullName)
+
+			} else {
+				requiresNewLogin := true
+
+				// if trying to log in an user that's not the currently logged in, try to reuse a previous token
+				if options.usernameProvided() {
+					glog.V(5).Infof("Checking existing authentication info for '%v'...\n", options.username)
+
+					for key, authInfo := range configStore.Config.AuthInfos {
+						glog.V(5).Infof("Checking %v...\n", key)
+
+						if key == options.username && len(authInfo.Token) > 0 { // only token for now
+							glog.V(5).Infof("Authentication exists for '%v', trying to use it...\n", options.username)
+
+							clientCfg.BearerToken = authInfo.Token
+							if userFullName, err := whoami(clientCfg); err == nil && userFullName == options.username {
+								err = config.UpdateConfigFile(userFullName, authInfo.Token, f.OpenShiftClientConfig.KClientConfig, configStore)
+								checkErr(err)
+								requiresNewLogin = false
+								fmt.Printf("Already logged into %v as '%v'\n", clientCfg.Host, userFullName)
+							} else {
+								glog.V(5).Infof("Token %v no longer valid for '%v', need to log in again\n", authInfo.Token, options.username)
+							}
+						}
+					}
+				}
+
+				// at this point we know we need to log in again
+				if requiresNewLogin {
+					clientCfg.BearerToken = ""
+					accessToken, err := tokencmd.RequestToken(clientCfg, os.Stdin, options.username, options.password)
+					checkErr(err)
+
+					if userFullName, err := whoami(clientCfg); err == nil {
+						err = config.UpdateConfigFile(userFullName, accessToken, f.OpenShiftClientConfig.KClientConfig, configStore)
+						checkErr(err)
+						fmt.Printf("Logged into %v as %v\n", clientCfg.Host, userFullName)
+					} else {
+						glog.Fatalf("%v\n", err)
+					}
+				}
+			}
+
+			// select a project to use
+			// if the user only has one project, use that one
+			// if user has multiple, take the first and print the others
+			// if --project were provided, use that one
+			oClient, err := client.New(clientCfg)
+			checkErr(err)
+			projects, err := oClient.Projects().List(labels.Everything(), labels.Everything())
+			checkErr(err)
+
+			if size := len(projects.Items); size > 0 {
+				if options.projectProvided() {
+					currentCtx := configStore.Config.Contexts[configStore.Config.CurrentContext]
+					for _, project := range projects.Items {
+						if project.Name == options.project {
+							currentCtx.Namespace = project.Name
+							config := configStore.Config
+							config.Contexts[config.CurrentContext] = currentCtx
+
+							if err = kclientcmd.WriteToFile(*configStore.Config, configStore.Path); err != nil {
```

Comment:
Check to see if there is an existing context you can use with the same server, creds, and namespace before overwriting one.